python/pyarrow/table.pxi

```diff
@@ -1263,6 +1270,28 @@ cdef class Table:
 
         return pyarrow_wrap_table(c_table)
 
+    def drop(self, columns):
+        """
+        Drop one or more columns and return a new table.
+        """
+        indices = []
+        for col in columns:
+            idx = self.schema.get_field_index(col)
+            if idx == -1:
+                raise KeyError("Column {!r} not found".format(col))
+            indices.append(idx)
+
+        indices.sort()
+        indices.reverse()
+
+        for i, idx in enumerate(indices):
+            if i == 0:
+                table = self.remove_column(idx)
```

Comment:
A nit: if you write `table = self` before the loop, you can dispense with this special case.

Reply:
nice, I love it!